File: 14-longest-common-prefix/longest-common-prefix.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs: 
            return ""
        
        prefix = ""
        instance = strs[0]

        for i in range(len(instance)):
            check = instance[i]
            for j in range(1, len(strs)):
                if i >= len(strs[j]) or strs[j][i] != check:
                    return prefix

            prefix += check

        return prefix

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string instance = strs[0];
        string prefix = "";
        for(int i=0;i<instance.size();++i) {
            char check = instance[i];
            for(int j=1;j<strs.size();++j) {
                if(i >= strs[j].size() || strs[j][i] != check) {
                    return prefix;
                }
            }
            prefix += check;
        
        }
        return prefix;
    }
};